### PR TITLE
Improve performance

### PR DESCRIPTION
**Pull request by gsanchezgavier**
- Makes cache write buff to be configurable.
- Makes cache update interval configurable.
- Launches updateIntances in the Cache in a non Locking way for the write buffer. 
- Removes Capacity Metric since is processing the envelopes of ValueMetrics twice and is not working properly in a multi instance environment.

### CODE REVIEW

**Review by bpeckNR — APPROVED** [review]
Looks good to me.

File: cfclient/cfapps/cache.go [hunk 1]
```diff
@@ -69,6 +68,22 @@ func (c *Cache) Start() {
...
+	c.isUpdating = true
+	var wg sync.WaitGroup
+	GetInstance().app.Log.Debug("Updating status of applications")
+	now := time.Now()
+	c.sync.RLock()
+	for _, v := range c.Collection {
+		wg.Add(1)
+		go v.UpdateInstancesWG(&wg)
+	}
+	c.sync.RUnlock()
```
Comment (davidbrota) [inline cfclient/cfapps/cache.go:81]: Could be there a problem if an app gets updated while it's being accessed from somewhere else? Here we release the lock while the update is still happening.

File: cfclient/cfapps/cache.go [hunk 2]
```diff
@@ -69,6 +68,22 @@ func (c *Cache) Start() {
 	}()
 }
 
+func (c *Cache) updateIntances() {
```
Comment (ardias) [inline cfclient/cfapps/cache.go]: small typo: updateIn**s**tances()

File: config/config.go [hunk 3]
```diff
@@ -74,6 +74,10 @@ func set() *Config {
 
 	// Cache purge threshold in minutes
 	v.SetDefault("FIREHOSE_CACHE_DURATION_MINS", 30)
+	// Cache instance update in seconds
+	v.SetDefault("FIREHOSE_CACHE_UPDATE_INTERVAL_SECS", 30)
```
Comment (davidbrota) [inline config/config.go]: shouldn't we increase the default value a little bit?

**Review by davidbrota — APPROVED** [review]
LGTM

File: cfclient/cfapps/cache.go [hunk 4]
```diff
@@ -32,7 +33,8 @@ func NewCache() *Cache {
 func (c *Cache) Start() {
 	go func() {
 		cacheDuration := app.Get().Config.GetDuration("FIREHOSE_CACHE_DURATION_MINS")
-		update := time.NewTicker(30 * time.Second).C
+		cacheUpdate := app.Get().Config.GetDuration("FIREHOSE_CACHE_UPDATE_INTERVAL_SECS")
```
Comment (ardias) [inline cfclient/cfapps/cache.go:36]: could you add a check here to make sure that users don't use a very low number and "destroy" their infrastructure ? I would always use the minimum between a default value (30 for example) and the value the user configured.

File: accumulators/capacity/capacity.go [hunk 5]
```diff
@@ -1,239 +0,0 @@
-// Copyright 2020 New Relic Corporation. All rights reserved.
```
Comment (paologallinaharbur) [inline accumulators/capacity/capacity.go:1]: There was a task to remove capacity in the backlog. My only doubt is if someone has some alerts on this

File: config/config.go (hunk 3 above)
Comment (gsanchezgavier) [reply config/config.go]: yes i will use 60 sec thanks. this should help on the api amounts of calls

File: cfclient/cfapps/cache.go [hunk 6]
```diff
@@ -15,13 +15,14 @@ type Cache struct {
...
 	sync        *sync.RWMutex
+	isUpdating  bool
 }
 
 // NewCache ...
 func NewCache() *Cache {
 	cache := &Cache{
 		Collection:  map[string]*CFApp{},
-		WriteBuffer: make(chan *CFApp, 1024),
+		WriteBuffer: make(chan *CFApp, app.Get().Config.GetDuration("FIREHOSE_CACHE_WRITE_BUFFER_SIZE")),
```
Comment (ardias) [inline cfclient/cfapps/cache.go:25]: As below, i would add a protection here for very low numbers. ex: min(1024,  FIREHOSE_CACHE_WRITE_BUFFER_SIZE)

**Review by ardias — CHANGES_REQUESTED** [review]

**Review by paologallinaharbur — APPROVED** [review]
LGMT

File: cfclient/cfapps/cache.go (hunk 4 above)
Comment (paologallinaharbur) [reply cfclient/cfapps/cache.go:36]: The tile config has a mimum defined as:

```
    constraints:
      min: 30
```

this should be enough, no? or you meant something different?

File: cfclient/cfapps/cache.go (hunk 6 above)
Comment (gsanchezgavier) [reply cfclient/cfapps/cache.go:25]: this protections are already at the tile configuration level

Comment (gsanchezgavier) [reply cfclient/cfapps/cache.go:25]: in the tile.yml configuration . I think this will prevents this

**Pull request merged**